aiida_common_workflows/workflows/relax/siesta/generator.py

```diff
@@ -0,0 +1,255 @@
+
+    with open(_filepath) as _thefile:
+        _protocols = yaml.full_load(_thefile)
+
+    _calc_types = {
+        'relaxation': {
+            'code_plugin':
+            'siesta.siesta',
+            'description':
+            'These are calculations used for the main and '
+            'only run of the code, computing the relaxation, in calc_engines the user must '
+            'define a code and options compatibles with plugin siesta.siesta'
+        }
+    }
+    _relax_types = {
+        RelaxType.ATOMS:
+        'the latice shape and volume is fixed, only the athomic positions are relaxed',
+        RelaxType.ATOMS_CELL:
+        'the lattice is relaxed together with the atomic coordinates. It allows'
+        'to target hydro-static pressures or arbitrary stress tensors.',
+        #    'constant_volume':'the cell volume is kept constant in a variable-cell relaxation: only'
+        #        'the cell shape and the atomic coordinates are allowed to change.  Note that'
+        #        'it does not make much sense to specify a target stress or pressure in this'
+        #        'case, except for anisotropic (traceless) stresses'
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Construct an instance of the inputs generator, validating the class attributes."""
+        super().__init__(*args, **kwargs)
+
+        def raise_invalid(message):
+            raise RuntimeError('invalid protocol registry `{}`: '.format(self.__class__.__name__) + message)
+
+        for k, v in self._protocols.items():  #pylint: disable=invalid-name
+
+            if 'parameters' not in v:
+                raise_invalid('protocol `{}` does not define the mandatory key `parameters`'.format(k))
+            if 'mesh-cutoff' in v['parameters']:
+                try:
+                    float(v['parameters']['mesh-cutoff'].split()[0])
+                    str(v['parameters']['mesh-cutoff'].split()[1])
+                except (ValueError, IndexError):
+                    raise_invalid(
+                        'Wrong format of `mesh-cutoff` in `parameters` of protocol '
+                        '`{}`. Value and units are required'.format(k)
+                    )
+
+            if 'basis' not in v:
+                raise_invalid('protocol `{}` does not define the mandatory key `basis`'.format(k))
+
+            if 'pseudo_family' not in v:
+                raise_invalid('protocol `{}` does not define the mandatory key `pseudo_family`'.format(k))
+            else:
+                famname = self._protocols[k]['pseudo_family']
+                try:
+                    Group.get(label=famname)
+                except exceptions.NotExistent:
+                    raise_invalid(
+                        'protocol `{}` requires `pseudo_family` with name {} '
+                        'but no family with this name is loaded in the database'.format(k, famname)
+                    )
+
+    #pylint: disable=too-many-locals
```

Comment:
By putting this statement here, it will actually disable it for the rest of the file. To only scope it to this method, place it after the end of the signature, so after the `):`